Sources/Core/AWSSDKIdentity/Sources/AWSSDKIdentity/IdentityProvidingClients/IdentityProvidingSTSClient.swift

```diff
@@ -0,0 +1,24 @@
+//
```

Comment:
This protocl is what allows dependency inversion. The actual `IdentityProvidingSTSClient` struct that conforms to the protocol here will be generated under `InternalAWSSTS`. Then, the auth scheme resolver of each public service module will construct an instance of the `IdentityProvidingSTSClient` struct and save it to auth option's `identityProperties`. Over in STS web identity cred resolver in `AWSSDKIdentity`, we just use the protocol to call the `getCredentialsWithWebIdentity`. This allows `AWSSDKIdentity` to use `InternalAWSSTS` without having direct dependency.